lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1647,35 +1652,112 @@ namespace Js
 
         return Js::JavascriptString::NewCopySz(strBuf, scriptContext);
 #else
-        // TODO (doilij): implement INTL_ICU version
-#ifdef INTL_ICU_DEBUG
-        Output::Print(_u("EntryIntl_FormatDateTime > returning null, fallback to JS\n"));
-#endif
-        return scriptContext->GetLibrary()->GetNull();
+        INTL_CHECK_ARGS(
+            args.Info.Count == 4 &&
+            DynamicObject::Is(args.Values[1]) &&
+            (TaggedInt::Is(args.Values[2]) || JavascriptNumber::Is(args.Values[2])) &&
+            JavascriptBoolean::Is(args.Values[3])
+        );
+
+        DynamicObject *state = DynamicObject::UnsafeFromVar(args.Values[1]);
+        double x = TaggedInt::Is(args.Values[1]) ? TaggedInt::ToDouble(args.Values[1]) : JavascriptNumber::GetValue(args.Values[1]);
```

Comment:
Looks like this shouldn't be checking `args.Values[1]`, since that is required to be a `DynamicObject` just above.